spatialdata/_core/_spatialdata.py

```diff
@@ -430,40 +450,59 @@ def add_image(
             See https://zarr.readthedocs.io/en/stable/api/storage.html for more details.
         overwrite
             If True, overwrite the element if it already exists.
-        _add_in_memory
+        _called_from_write
             Internal flag, to differentiate between an element added by the user and an element saved to disk by
             write method.
 
         Notes
         -----
         If the SpatialData object is backed by a Zarr storage, the image will be written to the Zarr storage.
         """
-        if _add_in_memory:
-            self._add_image_in_memory(name=name, image=image, overwrite=overwrite)
         if self.is_backed():
-            elem_group = self._init_add_element(name=name, element_type="images", overwrite=overwrite)
-            write_image(
-                image=self.images[name],
-                group=elem_group,
-                name=name,
-                storage_options=storage_options,
-            )
-            # reload the image from the Zarr storage so that now the element is lazy loaded, and most importantly,
-            # from the correct storage
+            des_path = os.path.join(str(self.path), "images", name)
```

Comment:
why first casting to `str` and then using join? I think it'd be nicer to have 
```
Path(self.path) / "images" / name
```

Reply:
and actually ideally self.path is already a `Path`

Reply:
I would also rename variable to `tgt_path`

Reply:
changed the name of the variable (and all instances of des/destination to tgt/target in `_spatialdata.py`). Also using `Path`.